src/Components/Browser.JS/src/Boot.Server.ts

```diff
@@ -89,13 +91,15 @@ async function boot(userOptions?: Partial<BlazorOptions>): Promise<void> {
 
   window['Blazor'].reconnect = reconnect;
 
-  const reconnectTask = reconnect(initialConnection);
+  await reconnect(initialConnection);
 
-  if (circuit) {
-    circuits.push(circuit);
-  }
+  // We render any additional component after all prerendered components have
+  // re-stablished the connection with the circuit.
+  const renderedComponents = await startCircuit(circuitId, initialConnection);
 
-  await reconnectTask;
+  if (!renderedComponents) {
+    logger.log(LogLevel.Information, 'No preregistered components to render.');
+  }
```

Comment:
It's very difficult to figure out by reading the PR diff how this is going to interact with reconnection cases. Do you feel we have adequate E2E (or if needed CTI) test coverage of reconnection?

Reply:
Not exactly sure what you mean here, but I did add a test that covers all possible cases:
* Only prerendered components.
* Only non-prerendered components.
* Mix of prerendered components + non-prerendered components.

Reply:
Sounds like the coverage is good then. I made that comment before I got to the end of the PR where the E2E tests were.